lib/python/rose/config_editor/window.py

```diff
@@ -406,6 +406,117 @@ def launch_open_dirname_dialog(self):
         open_dialog.destroy()
         return None
 
+    def launch_load_metadata_dialog(self):
+        """ Launches a dialoge for selecting a metadata path. """
+        open_dialog = gtk.FileChooserDialog(
+            title=rose.config_editor.DIALOG_TITLE_LOAD_METADATA,
+            action=gtk.FILE_CHOOSER_ACTION_SELECT_FOLDER,
+            buttons=(gtk.STOCK_CANCEL,
+                     gtk.RESPONSE_CANCEL,
+                     gtk.STOCK_OPEN,
+                     gtk.RESPONSE_OK))
+        open_dialog.set_transient_for(self.window)
+        open_dialog.set_icon(self.window.get_icon())
+        open_dialog.set_default_response(gtk.RESPONSE_OK)
+        response = open_dialog.run()
+        if response in [gtk.RESPONSE_OK, gtk.RESPONSE_ACCEPT,
+                        gtk.RESPONSE_YES]:
+            config_directory = open_dialog.get_filename()
+            open_dialog.destroy()
+            return config_directory
+        open_dialog.destroy()
+        return None
+
+    def launch_metadata_manager(self, paths):
+        """
+        Launches a dialogue where users may add or remove custom meta data
+        paths.
+        """
+        dialog = gtk.Dialog(
+            title=rose.config_editor.DIALOG_TITLE_MANAGE_METADATA,
+            buttons=(gtk.STOCK_CANCEL,
+                     gtk.RESPONSE_CANCEL,
+                     gtk.STOCK_OK,
+                     gtk.RESPONSE_OK)
+        )
+
+        class MetadataTable(object):
```

Comment:
This class can be declared by itself higher up the file rather than embedded in the function